metadata/metadata.go

```diff
@@ -219,12 +220,32 @@ func (err *ResourceExists) GRPCStatus() *status.Status {
 	return status.New(codes.AlreadyExists, err.Error())
 }
 
+type ResourceChanged struct {
```

Comment:
Should be ResourceChangedError

Reply:
None of the errors above do that lol. ResourceExists and ResourceNotFound don't have the Error suffix. Just following simba's code

Reply:
Its best practice https://gobyexample.com/errors 

Or move it into an errors.go file, or both

Reply:
I was just hoping my PR wouldn't be a big refactor of other pieces too. If I change mine I'll have to change the other ones as well. Usually consistency is more important than best practice (if you're already not following best practice). I'm more than happy to do it in another PR (you know how much I love refactoring for better practices).